docs: add explanation about publishing a slice (#149)
canonical/rockcraft: docs/how-to/publish-slice.rst

> How to make my slice definitions available for everyone?

Reviewer: ```suggestion
How to make custom slice definitions available for everyone?
```

********************************************************

> So now, you have created and tested your slice and want to make it **available for everyone.**

Reviewer: Could you please elaborate a bit more? This does tell people what to do, but with very little context.

Author: Done

How to make custom slice definitions available for everyone?
************************************************************

So right now, we have create a custom package slice and we have install it into a rocks.
Nevertheless, you saw in the previous tutorial that you need to call chisel into your rocks because your slice is not available for everyone.
In this tutorial we will see how to make it **accessible to everyone.**

#. Create a fork of this repository: https://github.com/canonical/chisel-releases
#. Create a branch and add your modification
#. Create a pull request
#. Wait for PR to be merged

> And that's it! You're custom slice definitions is available for everyone **Congrats!**

Reviewer: ```suggestion
And that's it! Your custom Chisel release and new slice definitions are now available in Chisel, and anyone can use them. **Congrats**! And thank you for your contribution.
```

How to make custom slice definitions available for everyone?
************************************************************

At this stage, you have created some package slice definitions and you have a custom Chisel release in your local development environment.
You have also tested this custom Chisel release, and it works! You believe there are others who could really use it as well, so **how can you make it accessible to everyone**?

It is as simple as proposing your changes into the upstream `Chisel releases repository <https://github.com/canonical/chisel-releases>`_:

> Create a fork of this repository: https://github.com/canonical/chisel-releases::

Reviewer: can you please bring the bullet list from before? you have nested ":" in this line now

    # Let's assume we are working with Ubuntu 22.04

>     git clone -b ubuntu-22.04 https://github.com/canonical/chisel-releases/

Reviewer: this is not a fork

Create a branch and add your modification::

    git checkout -b myopensslslice
    # make changes

>     git commit

Author: how it's doesn't work i don't understand

>     git push

Author: how it's doesn't work i don't understand

> After that create a pull request and wait for your pull request to be merged

Reviewer: ```suggestion
After that, create a pull request and wait for it to be merged.
```

How to make custom slice definitions available for everyone?
************************************************************

At this stage, you have created some package slice definitions and you have a custom Chisel release in your local development environment.
You have also tested this custom Chisel release, and it works! You believe there are others who could really use it as well, so **how can you make it accessible to everyone**?

It is as simple as proposing your changes into the upstream `Chisel releases repository <https://github.com/canonical/chisel-releases>`_:

#. Create a fork of https://github.com/canonical/chisel-releases and clone this repository:

    .. code-block:: bash

        # Let's assume we are working with Ubuntu 22.04
        $ git clone -b ubuntu-22.04 https://github.com/<your_github_username>/chisel-releases/

> #. Create a branch and add your modification:

Reviewer: ```suggestion
#. Add your modifications:

    .. code-block:: bash
        
        $ cp <path/to/your/slice/definitions/file> chisel-releases/slices

#. Create a new branch, commit and push it to your fork:
```

    .. code-block:: bash

>         $ git checkout -b myopensslslice

Reviewer: ```suggestion
        $ cd chisel-releases
        $ git checkout -b create-openssl-bins-slice
```

        # make changes
        $ git add .

>         $ git commit -m "Add slice definition file for 'name_of_the_package'"

Reviewer: ```suggestion
        $ git commit -m "feat: add new slice definitions for 'name_of_the_package'"
```

How to make custom slice definitions available for everyone?
************************************************************

At this stage, you have created some package slice definitions and you have a custom Chisel release in your local development environment.
You have also tested this custom Chisel release, and it works! You believe there are others who could really use it as well, so **how can you make it accessible to everyone**?

It is as simple as proposing your changes into the upstream `Chisel releases repository <https://github.com/canonical/chisel-releases>`_:

#. Fork this repository https://github.com/canonical/chisel-releases and clone your repository:

    .. code-block:: bash

        # Let's assume we are working with Ubuntu 22.04
        $ git clone -b ubuntu-22.04 https://github.com/<your_github_username>/chisel-releases.git

#. Create a branch:

    .. code-block:: bash

        $ cd chisel-releases
        $ git checkout -b create-openssl-bins-slice

> #. Add your modification:

Reviewer: ```suggestion
#. Add and commit your modifications:
```

    .. code-block:: bash

>         $ cp <path/to/your/slice/definitions/file> chisel-releases/slices

Reviewer: this will fail with the wrong path

.. _publishslice:

> How to make custom slice definitions available for everyone?

Reviewer 2: ```suggestion
How to make custom slice definitions available for everyone
```

Same as the install-slice how-to; to keep consistent with the create-slice one

************************************************************

At this stage, you have created some package slice definitions and you have a custom Chisel release in your local development environment.
You have also tested this custom Chisel release, and it works! You believe there are others who could really use it as well, so **how can you make it accessible to everyone**?

It is as simple as proposing your changes into the upstream `Chisel releases repository <https://github.com/canonical/chisel-releases>`_:

> #. Fork this repository https://github.com/canonical/chisel-releases and clone your repository:

Reviewer 2: ```suggestion
#. Fork this repository https://github.com/canonical/chisel-releases and clone your fork:
```
